dask/distributed/worker.py

```diff
@@ -371,6 +375,8 @@ def block(self):
         """
         self._listen_workers_thread.join()
         self._listen_scheduler_thread.join()
+        self._heartbeat_thread.event.set()  # stop heartbeat
```

Comment:
Check out the `close` method.

Block doesn't close things, it's a way to wait until things have been closed.